lib/Model/Binary.cpp

```diff
@@ -156,6 +156,16 @@ bool Binary::verify(VerifyHelper &VH) const {
       return VH.fail();
   }
 
+  // Make sure no segments overlap
+  if (!Segments().empty()) {
+    for (const auto &[LHS, RHS] :
+         llvm::zip(skip_back(Segments()), skip_front(Segments()))) {
```

Comment:
`:O` `:O` `:O` `:O`

Reply:
It's your idea actually. We had the same construct in the ABI PR, which was super easy to adopt here as well.

P. S. We probably should also introduce a named helper for this, something like
```cpp
template<typename Range>
constexpr auto pairwise_view(Range R) { return llvm::zip(skip_back(R), skip_front(R)); }
```